### PR TITLE
[CP] Migrate command usage values (#139383)

### PR DESCRIPTION
**Pull request by eliasyishak**
Fixes:
- https://github.com/flutter/flutter/issues/141017

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/gradle.dart [hunk 1]
```diff
@@ -640,10 +640,9 @@ class AndroidGradleBuilder implements AndroidBuilder {
     required Directory outputDirectory,
     required String buildNumber,
   }) async {
-
     final FlutterManifest manifest = project.manifest;
-    if (!manifest.isModule && !manifest.isPlugin) {
-      throwToolExit('AARs can only be built for plugin or module projects.');
+    if (!manifest.isModule) {
+      throwToolExit('AARs can only be built for module projects.');
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/android/gradle.dart:645]: is this related?

File: packages/flutter_tools/lib/src/commands/build_aar.dart [hunk 2]
```diff
@@ -102,6 +123,14 @@ class BuildAarCommand extends BuildSubCommand {
       'This command assumes that the entrypoint is "lib/main.dart". '
       'This cannot currently be configured.';
 
+  @override
+  Future<void> validateCommand() async {
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/commands/build_aar.dart:127]: same question, is this required for usage values?

File: packages/flutter_tools/lib/src/android/gradle.dart (hunk 1 above)
Comment (eliasyishak) [reply packages/flutter_tools/lib/src/android/gradle.dart:645]: No, this was brought in from your [commit](https://github.com/flutter/flutter/pull/141019/commits/d2bd4441c0c6330a6729c0a5c37b3a68a4efdb50) actually. There was a change you made that my CP was depending on. I thought originally that I would need to only merge previously committed PRs for a CP.

It's probably best to revert the changes from your commit and just make the necessary changes for this PR to work right?

File: packages/flutter_tools/lib/src/commands/build_aar.dart (hunk 2 above)
Comment (eliasyishak) [reply packages/flutter_tools/lib/src/commands/build_aar.dart]: It is not required for this PR

Comment (eliasyishak) [reply packages/flutter_tools/lib/src/commands/build_aar.dart:128]: Well turns out this was necessary for this [test](https://github.com/eliasyishak/flutter/blob/cp-migrate-usage-values-ga4/packages/flutter_tools/test/commands.shard/hermetic/build_aar_test.dart#L70-L73) 🙃

**Comment by eliasyishak** [comment]
@christopherfujino looks like this is ready for your review now

File: packages/flutter_tools/lib/src/commands/build_aar.dart (hunk 2 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/commands/build_aar.dart:128]: Let's delete the test rather than change the behavior of the tool in CP.

File: packages/flutter_tools/test/commands.shard/hermetic/build_aar_test.dart [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      logger: logger,
+      processManager: processManager,
+    );
+    fakeAnalytics = getInitializedFakeAnalyticsInstance(
+      fs: fs,
+      fakeFlutterVersion: FakeFlutterVersion(),
+    );
+  });
+
+  testUsingContext('will not build an AAR for a plugin', () async {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/build_aar_test.dart:51]: If the tested behavior did not already exist on the 3.16 branch, let's just delete this test, per https://github.com/flutter/flutter/pull/141019/files#r1445425246

**Review by christopherfujino — APPROVED** [review]
This LGTM once you remove the `BuildAarCommand.validateCommand()` override and the failing test.
